### ISSUE TITLE
styleItalics undeclared identifier

### ISSUE THREAD

**Issue by ghost** [icyphox/fab#11]
affecting line is (47, 19)

**Comment by icyphox** [comment]
oh boy, i really should build these against the latest Nim releases more often. looks like we have a regression here.

**Comment by ghost** [comment]
I would've looked through nim's source but was being kinda lazy and just commented out the italics part lol :p

**Comment by ghost** [comment]
Lines 499-512 terminal.nim

seems like they forgot to add it to the proc
```
proc setStyle*(f: File, style: set[Style]) =
  ## Sets the terminal style.
  when defined(windows):
    let h = conHandle(f)
    var old = getAttributes(h) and (FOREGROUND_RGB or BACKGROUND_RGB)
    var a = 0'i16
    if styleBright in style: a = a or int16(FOREGROUND_INTENSITY)
    if styleBlink in style: a = a or int16(BACKGROUND_INTENSITY)
    if styleReverse in style: a = a or 0x4000'i16 # COMMON_LVB_REVERSE_VIDEO
    if styleUnderscore in style: a = a or 0x8000'i16 # COMMON_LVB_UNDERSCORE
    discard setConsoleTextAttribute(h, old or a)
  else:
    for s in items(style):
      f.write(ansiStyleCode(s))
```

**Comment by icyphox** [comment]
oh great. you want to submit a patch?

**Comment by ghost** [comment]
Once I figure out how to do it like nim, currently I'm looking at how they add italics in C/C++, but it seems to be widely unsupported for many terminal types?

**Comment by icyphox** [comment]
that’s correct. italics aren’t on most terminals. but it doesn’t break anything afaik, just reverts to inverted colors.

**Comment by ghost** [comment]
I'll keep on looking but so far notta. this is the only version of Fab I've used, was it working before?

**Comment by icyphox** [comment]
yep it was. but idk when it broke.

**Comment by ghost** [comment]
Not sure how this would apply to Nim directly, seems like a dirty work-around

"\e[3mitalic\e[0m"

**Comment by icyphox** [comment]
but that’s not cross-platform. that won’t work on non-POSIX systems.

**Comment by ghost** [comment]
As of now I pushed a patch to terminal.nim just commenting out the affecting line and explained the issue at hand.

Maybe we'll get a fix soon.

**Comment by kaushalmodi** [comment]
@blmvxer What is your Nim version?

**Comment by ghost** [comment]
0.18.0-1 arch Linux, and same for Windows 10

**Comment by kaushalmodi** [comment]
> 0.18.0

I thought so .. you should update to Nim 0.19.4, the latest stable version.

@icyphox You should update the minimum required Nim for this nimble package to 0.19.0.

**Comment by kaushalmodi** [comment]
@icyphox See my earlier comment in https://github.com/icyphox/fab/pull/4#issue-195528305

> Now the `styleItalic` and `styleStrikethrough` values of `Style` enum (Nim 0.18.1+) are used.

Nim 0.18.1 was the `devel` version then.. these changes in that devel got released in Nim stable release 0.19.0.

> Without that, you will get a compile error. 

The kind @blmvxer is seeing right now :)

**Comment by icyphox** [comment]
@kaushalmodi right, yeah. i’m on 0.19.0 at the moment. i’ll upgrade the minimum version required to that, in this package.
